src/EnergyPlus/CurveManager.cc

```diff
@@ -2547,6 +2554,18 @@ namespace CurveManager {
 				}
 			}
 
+			IndVarSwitch = false;
+			if ( SameString( Alphas( 4 ), "WAVELENGTH" ) && SameString( Alphas( 5 ), "ANGLE" ) ) {
+				IndVarSwitch = true;
```

Comment:
Is it really worth all the trouble to flip the variables?  The unit type fields are not required, so you can't be sure these will be input.  And, in general, when a component model uses a curve/table object, it expects X1 to be a specific variable (e.g. Outdoor dry bulb temperature).  Seems it would be better to enforce X1 is always angle and X2 is always wavelength (or the opposite) and in the function that calls this, check the max/min values for X1 and X2 and throw a warning if they don't line up with the proper variable type.